primitives/runtime/src/lib.rs

```diff
@@ -395,17 +395,37 @@ impl Size for PreComputedSize {
 /// Note: If the same `bridge_id` is used across different chains (for example, if one source chain
 /// is bridged to multiple target chains), then all the derived accounts would be the same across
 /// the different chains. This could negatively impact users' privacy across chains.
-pub fn derive_account_id<AccountId>(bridge_id: ChainId, id: SourceAccount<AccountId>) -> H256
+pub fn derive_account_id<AccountId>(bridge_id: ChainId, id: SourceAccount<AccountId>) -> H160
 where
 	AccountId: Encode,
 {
 	match id {
-		SourceAccount::Root =>
-			(ROOT_ACCOUNT_DERIVATION_PREFIX, bridge_id).using_encoded(blake2_256),
-		SourceAccount::Account(id) =>
-			(ACCOUNT_DERIVATION_PREFIX, bridge_id, id).using_encoded(blake2_256),
+		SourceAccount::Root => {
+			let h256 = (ROOT_ACCOUNT_DERIVATION_PREFIX, bridge_id).using_encoded(blake2_256);
+			H160::from_slice(&h256[0..20])
+		},
+		SourceAccount::Account(id) => {
+			// convert id to h160
+			let source_h160 = H160::from_slice(&id.encode()[0..20]);
```

Comment:
This might panic since the `AccountId` is a generic param, the true type might be a u64 or something else in the test cases.